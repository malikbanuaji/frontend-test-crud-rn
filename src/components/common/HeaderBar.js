import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Spacing} from '../../styles';
import HeaderLoading from './HeaderLoading';

export default function HeaderBar({children, isLoading, loadingMessage}) {
  return (
    <View testID={'header-bar'} style={styles.headerBar}>
      {isLoading ? <HeaderLoading text={loadingMessage} /> : children}
    </View>
  );
}

const styles = StyleSheet.create({
  headerBar: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: Spacing.base,
    height: 72,
  },
});
