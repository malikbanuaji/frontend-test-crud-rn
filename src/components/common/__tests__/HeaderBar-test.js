import {render} from '@testing-library/react-native';
import React from 'react';
import {Text, View} from 'react-native';
import renderer from 'react-test-renderer';
import HeaderBar from '../HeaderBar';

test('renders correctly', () => {
  const tree = render(<HeaderBar />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly when loading', () => {
  const tree = render(
    <HeaderBar isLoading loadingMessage={'Loading...'} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should renders with 3 children', () => {
  const {getByTestId} = render(
    <HeaderBar isLoading={false} loadingMessage={'Loading...'}>
      <Text children={'John'} />
      <Text children={'Doe'} />
      <Text children={'Bar'} />
    </HeaderBar>,
  );
  const headerBar = getByTestId('header-bar');
  expect(headerBar.children.length).toBe(3);
});
