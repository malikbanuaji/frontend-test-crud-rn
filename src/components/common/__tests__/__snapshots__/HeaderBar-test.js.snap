// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": 72,
      "padding": 16,
    }
  }
  testID="header-bar"
/>
`;

exports[`renders correctly when loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": 72,
      "padding": 16,
    }
  }
  testID="header-bar"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#93b5e1"
        hidesWhenStopped={true}
        size={30}
        style={
          Object {
            "marginRight": 8,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 14,
          }
        }
      >
        Loading...
      </Text>
    </View>
  </View>
</View>
`;
